### PR TITLE
Merge pouchdb-req-http-query.

### PR DESCRIPTION
**Pull request by loic**

### CODE REVIEW

**Comment by loic** [comment]
cc @gr2m @marten-de-vries

**Comment by marten-de-vries** [comment]
Nice work. The indents in package.json are looking weird in the diff (both spaces & tabs in use, perhaps?), but otherwise this is ready for merging when green.

**Comment by loic** [comment]
Dunno what happened with those indents hehe, fixed it now. Travis has been stuck since yesterday on all PRs, I wonder if the pouchdb org is rate limited or something, it seems to happen often and I don't have this problem on my fork.

**Comment by marten-de-vries** [comment]
Thanks for the fix. Indeed, Travis rate limits per organization. And Pouchdb's jobs are pretty large. Also, Travis had problems today: https://www.traviscistatus.com/incidents/c1s4dlyxd4lf

File: packages/node_modules/pouchdb-req-http-query/lib/index.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    xhr.open(req.method, url, true);
+    for (var name in headers) {
+      if (headers.hasOwnProperty(name)) {
+        if (xhr.setDisableHeaderCheck) {
+          xhr.setDisableHeaderCheck(true);
+        }
+        xhr.setRequestHeader(name, headers[name]);
+      }
+    }
+    xhr.send(req.body === "undefined" ? null : req.body);
```
Comment (loic) [inline packages/node_modules/pouchdb-req-http-query/lib/index.js]: Could this be missing a `typeof`?

Comment (marten-de-vries) [reply packages/node_modules/pouchdb-req-http-query/lib/index.js]: I think it's correct as-is: http://docs.couchdb.org/en/2.0.0/json-structure.html#request-object

Comment (loic) [reply packages/node_modules/pouchdb-req-http-query/lib/index.js]: I see! It's probably worth a comment with a link to that doc, so someone doesn't "helpfully" adds a `typoeof` 😛

Comment (marten-de-vries) [reply packages/node_modules/pouchdb-req-http-query/lib/index.js]: Good point. Will do so when merging.

**Comment by marten-de-vries** [comment]
Merged: b26f66947fb759c45b181affe22047b7dcdb6d1d

**Pull request merged**